### ISSUE TITLE
Tessellating block in world

### ISSUE THREAD

**Issue by Szedann** [copycats-plus/copycats#63]
https://mclo.gs/JY49mK7#L3303

**Comment by Bennyboy1695** [comment]
What caused this issue? Was it loading the world after the update or inserting a specific block into the block?

**Comment by Szedann** [comment]
it crashed during gameplay after an update, presumably when loading a chunk

**Comment by Bennyboy1695** [comment]
Was this a world with blocks that changed to multistates? 
And if so did you enable to config option to convert them?

**Comment by Szedann** [comment]
yes it was, I did not know there was a config allowing you to do that so no
I'd say that should probably be enabled by default

**Comment by Bennyboy1695** [comment]
try enabling that option and reloading the world and see if that crash happens again

**Comment by Szedann** [comment]
https://mclo.gs/fyVXXJL#L3330 once again, after enabling

**Comment by Bennyboy1695** [comment]
Do you know what mod or whatever is using continuity? As thats the mod causing the issue but testing it by itself isnt causing me to crash

**Comment by Szedann** [comment]
not entirely sure, but most copycat slabs work fine

**Comment by IThundxr** [comment]
that's just continuity wrapping models, not continuities fault here, you shouldn't be relying on a hash map here if your using one and instead use something like a Concurrent hash map or a map wrapped in Collections#synchronizedMap

**Comment by IThundxr** [comment]
actually you can probably get away with storing a new arraylist with the contents of materials.entrySet()

**Comment by Bennyboy1695** [comment]
Please try the version just released on CF/Modrinth and if the crash still occurs please re-open the issue and we will try another fix

**Comment by Szedann** [comment]
the issue persists
https://mclo.gs/pW4Euym

**Comment by Bennyboy1695** [comment]
We will try the other suggestion by IThundxr. If that also doesn't help then i will leave it up to Lysine as they are our rendering god in the first place XD

**Comment by Bennyboy1695** [comment]
Please try this one and see if it still happens - https://drive.google.com/file/d/1bLfHQUEXEsn9Y9gvmJ3Iv-8OaEQbrZ_C/view?usp=drivesdk
Posted it on my google drive for CF/Modrinth so that i dont deploy a build that doesnt actually fix anything

**Comment by Szedann** [comment]
the file needs permission to view, could you make it public?

**Comment by Bennyboy1695** [comment]
thats odd should have been public 😂 should be now though

**Comment by Szedann** [comment]
unless the fixed version needs to be on the server too, it did not fix the issue

[crash log](https://mclo.gs/DCZ8EHc)

**Comment by hlysine** [comment]
[UNZIP ME Copycats-fabric.1.20.1-1.3.4-TEST.9339880.zip](https://github.com/user-attachments/files/15935106/UNZIP.ME.Copycats-fabric.1.20.1-1.3.4-TEST.9339880.zip)

Please try this version on both the server and clients. It's a client-only fix, but since it also includes some other fixes, it's better to install on the server as well to be safe.

**Comment by hlysine** [comment]
Newer versions have completely reworked internals so this issue no longer applies.
